enos/enos.py

```diff
@@ -220,92 +213,90 @@ def install_os(env=None, **kwargs):
-    for flavor in default_flavors:
-        if flavor[0] not in current_flavors:
-            nova.flavors.create(name=flavor[0],
-                        ram=flavor[1],
-                        disk=flavor[2],
-                        vcpus=flavor[3])
-            logging.info("%s has been created on OpenStack" % flavor[0])
-
-    # Install default network
-    neutron = ntnclient.Client('2', session=sess)
-    network_name = 'public1'
-    network_id = ''
-    networks = neutron.list_networks()['networks']
-    if network_name not in map(itemgetter('name'), networks):
-        network = {'name': network_name,
-                   'provider:network_type': 'flat',
-                   'provider:physical_network': 'physnet1',
-                   'router:external': True
-        }
-        res = neutron.create_network({'network': network})
-        network_id = res['network']['id']
-        logging.info("%s network has been created on OpenStack" % network_name)
-
-    if not network_id:
-        logging.error("no network_id for %s network" % network_name)
-        sys.exit(32)
-
-    # Install default subnet
-    subnet_name = '1-subnet'
-    subnets = neutron.list_subnets()['subnets']
-    if subnet_name not in map(itemgetter('name'), subnets):
-        subnet = {'name': subnet_name,
-                  'network_id': network_id,
-                  'cidr': '10.0.2.0/24',
-                  'ip_version': 4}
-        neutron.create_subnet({'subnet': subnet})
-        logging.info("%s has been created on OpenStack" % subnet_name)
+    for flavor in flavors:
+        cmd.append('openstack flavor create %s' \
+                ' --id auto' \
+                ' --ram %s' \
+                ' --disk %s' \
+                ' --vcpus %s' \
+                ' --public' % (flavor[0], flavor[1], flavor[2], flavor[3]))
+
+    # security groups - allow everything 
+    protos = ['icmp', 'tcp', 'udp']
+    for proto in protos:
+        cmd.append('openstack security group rule create default' \
+                ' --protocol %s' \
+                ' --dst-port 1:65535' \
+                ' --src-ip 0.0.0.0/0' % proto)
+    
+    # quotas - set some unlimited for admin project
+    quotas = ['cores', 'ram', 'instances']
+    for quota in quotas:
+        cmd.append('nova quota-class-update --%s -1 default' % quota)
+
+    quotas = ['fixed-ips', 'floating-ips']
+    for quota in quotas:
+        cmd.append('openstack quota set --%s -1 admin' % quota)
+
+    # default network (one public / one provite)
+    cmd.append('openstack network create public' \
+            ' --share' \
+            ' --provider-physical-network physnet1' \
+            ' --provider-network-type flat' \
+            ' --external')
+    cmd.append('openstack subnet create public-subnet' \
+            ' --network public' \
+            ' --subnet-range %s' \
+            ' --no-dhcp'
+            ' --allocation-pool start=%s,end=%s' \
+            ' --gateway %s' \
```

Comment:
Yep I guess we can specify the DNS as well for the public network.
could you give me more context : are you putting vm in the private or public network ?